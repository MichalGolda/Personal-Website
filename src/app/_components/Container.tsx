export type ContainerProps = React.PropsWithChildren<{
  className?: string;
}>;

const Container: React.FC<ContainerProps> = ({ className, children }) => {
  return (
    <div className={`max-w-[1320px] mx-auto ${className}`}>{children}</div>
  );
};

export default Container;
