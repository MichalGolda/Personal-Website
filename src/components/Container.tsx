import styled from "styled-components";

const Container = styled.div`
  width: 100%;
  padding-left: 16px;
  padding-right: 16px;

  @media (${({ theme }) => theme.mediaQuery.small}) {
    padding-left: 32px;
    padding-right: 32px;
  }

  @media (${({ theme }) => theme.mediaQuery.sm}) {
    padding-left: 64px;
    padding-right: 64px;
  }

  @media (${({ theme }) => theme.mediaQuery.md}) {
    margin: 0 auto;
    max-width: 1024px;
  }
`;

export default Container;
