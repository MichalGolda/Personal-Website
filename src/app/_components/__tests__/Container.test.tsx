import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Container from "../Container";

describe("Container component", () => {
  it("renders without errors", () => {
    const { container: root } = render(
      <Container className="test">
        <div data-testid="children"></div>
      </Container>
    );

    const container = root.firstChild;

    expect(container).toBeInTheDocument();
    expect(container).toHaveClass("test");

    const children = screen.getByTestId("children");

    expect(children).toBeInTheDocument();
  });

  it("renders unchanged", () => {
    const { container } = render(
      <Container className="test">
        <div></div>
      </Container>
    );

    expect(container).toMatchSnapshot();
  });
});
