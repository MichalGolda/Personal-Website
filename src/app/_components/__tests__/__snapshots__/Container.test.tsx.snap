// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component renders unchanged 1`] = `
<div>
  <div
    class="md:max-w-[1024px] lg:max-w-[1320px] w-full mx-auto md:px-16 px-4 test"
  >
    <div />
  </div>
</div>
`;
